File: src/main/java/com/example/transactions/application/use_cases/GetStatisticService.java
package com.example.transactions.application.use_cases;

import com.example.transactions.application.ports.in.GetStatisticUseCase;
import com.example.transactions.application.ports.out.TransactionStorage;
import com.example.transactions.domain.Statistic;
import com.example.transactions.domain.Transaction;
import java.util.List;
import org.springframework.stereotype.Service;

@Service
public class GetStatisticService implements GetStatisticUseCase {

    private final TransactionStorage transactionStorage;

    public GetStatisticService(TransactionStorage transactionStorage) {
        this.transactionStorage = transactionStorage;
    }

    @Override
    public Statistic calculate() {
        List<Transaction> transactions = transactionStorage.getLastMinute();
        return Statistic.generate(transactions);
    }
}

File: application/src/main/kotlin/com/example/application/use_cases/GetStatisticService.kt
package com.example.application.use_cases

import com.example.application.dto.StatisticResponse
import com.example.application.ports.`in`.GetStatisticUseCase
import com.example.application.ports.out.TransactionStorage
import com.example.domain.Statistic
import com.example.domain.Transaction
import org.springframework.stereotype.Service

@Service
class GetStatisticService(private val transactionStorage: TransactionStorage) : GetStatisticUseCase {

    override fun calculate(): StatisticResponse {
        val transactions: List<Transaction> = transactionStorage.getLastMinutes()
        val statistic: Statistic = Statistic.generate(transactions)
        return StatisticResponse(statistic)
    }
}
